helix-view/src/document.rs

```diff
@@ -291,8 +378,58 @@ impl Document {
         }
     }
 
+    pub(crate) async fn save_impl<'a>(
+        text: &'a Rope,
+        writer: &'a mut (impl tokio::io::AsyncWriteExt + Unpin),
+        encoder: &'a mut encoding_rs::Encoder,
+    ) -> Result<(), anyhow::Error> {
+        let iter = text.chunks();
+        let iter_len = iter.clone().count();
```

Comment:
We don't need `iter_len` (which I think I mentioned on matrix, but probably after you wrote this).  It's O(N) to compute.

In my comments below I'll explain what we actually want to do.